### PR TITLE
Update K8s minimum version supported by Rook from 1.7 to 1.8

### LINKED ISSUE #2040: Update K8s minimum version supported by Rook from 1.7 to 1.8
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
Given the dwindling user base on 1.7 especially in the timeframe that Rook 0.9 would be released, there is not a compelling reason to keep supporting 1.7. There is some support/dev cost to maintaining each release. As an example, #1995 would require test code to workaround a feature that is not available in 1.7.

At the same time as this change, the pre-req page should be updated with the notes from a recent community discussion on the min version that rook supports.
```
We are not keeping with the client-go matrix anymore since we don’t support older than 1.7 and that’s OK. Our docs do not state a policy to keep with client-go.  At a minimum, the newest 3-4 versions of k8s should be supported and when we consider dropping a version it needs to be discussed and rationalized for a good reason with expected minimal impact to our users.
```

### PR DESCRIPTION
**Pull request by rtnpro**
**Description of your changes:**

Update k8s minimum supported version from 1.7 to 1.8. As a result, we drop the integration tests for k8s version 1.7 for upcoming 0.9 release.

**Which issue is resolved by this Pull Request:**
Resolves #2040 

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

[skip ci]

### CODE REVIEW

File: Documentation/upgrade.md [hunk 1]
```diff
@@ -410,7 +410,7 @@ kubectl -n rook get pod -l app=rook-ceph-mds -o jsonpath='{range .items[*]}{.met
 At this point, your Rook cluster should be fully upgraded to running version `rook/ceph:v0.8.0` and the cluster should be healthy according to the steps in the [health verification section](#health-verification).
 
 ## Upgrading Kubernetes
```
Comment (travisn) [inline Documentation/upgrade.md]: You could actually remove this whole section. We haven't supported TPRs for a couple releases.

File: tests/README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ You can choose any Kubernetes flavor of your choice.  The test framework only de
 The framework also provides scripts to install Kubernetes. There are two scripts to start the cluster:
 1. **Minikube** (recommended for MacOS): Run [minikube.sh](/tests/scripts/minikube.sh) to setup a single-node Minikube Kubernetes.
     1. Minikube v0.21.0 and higher is supported. Older minikube versions do not have cephfs or rbd tools installed.
-    1. If using Kubernetes v1.7.3 or higher, then Minikube v0.22.2 or higher is required.
+    1. If using Kubqernetes v1.8 or higher, then Minikube v0.23.0 or higher is required.
```
Comment (travisn) [inline tests/README.md]: typo: kubqernetes

**Comment by travisn** [comment]
@jbw976 thanks for the reminder. Given how simple the change is to the jenkinsfile and everything else is pretty much doc/script changes that are not in the build or integration tests, we can add the `[skip ci]` tag

**Comment by travisn** [comment]
The content LGTM. As the last step, could you squash some of the commits? Perhaps we could have one commit for the doc changes and one commit for the jenkinsfile/script changes. Thanks!

**Comment by rtnpro** [comment]
@travisn I squashed this branch into 2 commits:
- for script changes
- another for doc update.

File: PendingReleaseNotes.md [hunk 3]
```diff
@@ -9,6 +9,7 @@
 - Ceph OSDs will be automatically updated by the operator when there is a change to the operator version or when the OSD configuration changes. See the [OSD upgrade notes](Documentation/upgrade-patch.md#object-storage-daemons-osds).
 - Rook Ceph block storage provisioner can now correctly create erasure coded block images. See [Advanced Example: Erasure Coded Block Storage](Documentation/block.md#advanced-example-erasure-coded-block-storage) for an example usage.
 - [Network File System (NFS)](https://github.com/nfs-ganesha/nfs-ganesha/wiki) is now supported by Rook with a new operator to deploy and manage this widely used server. NFS servers can be automatically deployed by creating an instance of the new `nfsservers.nfs.rook.io` custom resource. See the [NFS server user guide](Documentation/nfs.md) to get started with NFS.
+- The minimum version of Kubernetes supported by Rook changed from `1.7` to `1.8`.
```
Comment (jbw976) [inline PendingReleaseNotes.md]: I would consider referencing this in the breaking changes section too.  In that section, you can mention that users of 1.7 should consider upgrading their k8s cluster (maybe even link to docs to help with that if you can find a good link?).

Comment (rtnpro) [reply PendingReleaseNotes.md]: @jbw976 Update docs as per your comment.

Comment (jbw976) [reply PendingReleaseNotes.md]: looks great, thanks!

**Comment by rtnpro** [comment]
@travisn can we merge this? Then I will rebase my other PR for supporting Storageclass reclaim policy on top of latest master branch.

**Comment by travisn** [comment]
@rtnpro yes, it's ready to merge, but jenkins seems to be having an issue that is preventing it from being marked green. investigating...

**Pull request merged**